utilities/oadp.py

```diff
@@ -0,0 +1,160 @@
+import logging
+from contextlib import contextmanager
+
+from ocp_resources.backup import Backup
+from ocp_resources.datavolume import DataVolume
+from ocp_resources.storage_profile import StorageProfile
+from ocp_resources.virtual_machine import VirtualMachine
+
+from utilities import console
+from utilities.constants import (
+    ADP_NAMESPACE,
+    FILE_NAME_FOR_BACKUP,
+    LS_COMMAND,
+    OS_FLAVOR_RHEL,
+    TEXT_TO_TEST,
+    TIMEOUT_5MIN,
+    TIMEOUT_20SEC,
+    Images,
+)
+from utilities.infra import (
+    cleanup_artifactory_secret_and_config_map,
+    get_artifactory_config_map,
+    get_artifactory_secret,
+    get_http_image_url,
+    get_pod_by_name_prefix,
+    unique_name,
+)
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+def delete_velero_resource(resource, client):
+    velero_pod = get_pod_by_name_prefix(dyn_client=client, pod_prefix="velero", namespace=ADP_NAMESPACE)
+    command = ["./velero", "delete", resource.kind.lower(), resource.name, "--confirm"]
+    velero_pod.execute(command=command)
+
+
+class VeleroBackup(Backup):
+    def __init__(
+        self,
+        name,
+        namespace=ADP_NAMESPACE,
+        included_namespaces=None,
+        client=None,
+        teardown=False,
+        yaml_file=None,
+        excluded_resources=None,
+        wait_complete=True,
+        snapshot_move_data=False,
+        storage_location=None,
+        timeout=TIMEOUT_5MIN,
+        **kwargs,
+    ):
+        super().__init__(
+            name=unique_name(name=name),
+            namespace=namespace,
+            included_namespaces=included_namespaces,
+            client=client,
+            teardown=teardown,
+            yaml_file=yaml_file,
+            excluded_resources=excluded_resources,
+            storage_location=storage_location,
+            snapshot_move_data=snapshot_move_data,
+            **kwargs,
+        )
+        self.wait_complete = wait_complete
+        self.timeout = timeout
+
+    def __enter__(self):
+        super().__enter__()
+        if self.wait_complete:
+            self.wait_for_status(
+                status=self.Status.COMPLETED,
+                timeout=self.timeout,
+            )
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        delete_velero_resource(resource=self, client=self.client)
+        super().__exit__(exception_type, exception_value, traceback)
+
+
+@contextmanager
+def create_rhel_vm(
+    storage_class,
+    namespace,
+    dv_name,
+    vm_name,
+    rhel_image,
+    client=None,
+    wait_running=True,
+    volume_mode=None,
+):
+    artifactory_secret = get_artifactory_secret(namespace=namespace)
+    artifactory_config_map = get_artifactory_config_map(namespace=namespace)
+    try:
```

Comment:
Why do we need `try` here?

Reply:
this is not the way to handle it becuse it means that any function suing the cm and secret would have to implement it. should be done in `get_artifactory_secret` and `get_artifactory_config_map`  (in another PR)